system/include/emscripten/heap.h

```diff
@@ -31,10 +32,36 @@ uintptr_t *emscripten_get_sbrk_ptr(void);
 // cannot be used to shrink the size of the heap.
 int emscripten_resize_heap(size_t requested_size) EM_IMPORT(emscripten_resize_heap);
 
-// Returns the current size of the WebAssembly heap.
+// Returns the base address of the global data section. The global data section
+// contains the constant global and static function local data, either zero- or
+// non-zero initialized. This address stays constant throughout program lifetime.
+uintptr_t emscripten_get_global_base(void);
+
+// Returns the end address of the global data section. N.b. the memory location
+// pointed to by this end value is not part of the global data section itself, i.e.
+// the global data section consists of the half-open range [global_base,
+// global_end[ in the usual begin-end pointer span style. This address stays
+// constant throughout program lifetime.
+uintptr_t emscripten_get_global_end(void);
+
+// Returns the starting address of the dynamic memory heap. This value corresponds
+// to the the sbrk value at program startup, when no dynamic memory allocations
+// have yet been made. This address stays constant throughout program lifetime.
+uintptr_t emscripten_get_dyn_heap_base(void);
```

Comment:
I've been working on a change to try to clean up some of the `heap` vs `memory` confusion by the way: https://github.com/emscripten-core/emscripten/pull/18177.

If/when that lands perhaps we don't need the "dyn_" prefix here?

Reply:
Sorry, stalled on this PR, now getting back to tending to my PRs.

I'm now thinking maybe `emscripten_get_dynamic_heap_base()` might be a better name here, i.e. just write it out in full? Or any other suggestions?